### DISCUSSION TITLE
How to test the hooks generated by @wagmi/cli, specifically that a usePrepareContractWrite returns a valid config

### DISCUSSION THREAD

**Discussion by hardingjam** [wevm/wagmi#3280] [Q&A]
React project using Vite, Rainbowkit and Wagmi.
I have used @wagmi/cli to create wrapper hooks for various contract functions (eg, usePrepareCustomContractWrite), and created my own wrapped "useCreate" that wraps the prepare and write hooks from generated.ts, and returns their various states

I don't need to test the generated useCustomContractWrite hook, but I want to test that usePrepareCustomContractWrite returns a config object only when it receives a valid constructionConfig.

```
  const {
        config: prepareContractWriteConfig,
        isLoading: prepareContractWriteIsLoading,
        isSuccess: prepareContractWriteSuccess,
        error: prepareContractWriteError,
    } = usePrepareCustomContactWrite({
        args: [constructionConfig],
    });

```

I am using Vitest and I want to do something akin to:

```
describe("useCreate", () => {
    it("returns a valid configuration when given acceptable arguments", async () => {
        const acceptableArgs = {
            adminLedger: "0xValidAddress",
            name: "ValidName",
        };
        const { result, waitFor } = renderHook(() => useCreate(acceptableArgs));
        await waitFor(() => {
            // Assuming useCreate exposes the config from usePrepareCustomContactWrite
            expect(result.current.prepareContractWriteConfig).toBeDefined();
        });
    });
});

```

In this test - the waitFor times out every time. Is there a simple example of a mocked connector and wagmiConfig that I can use in these tests? The [tests in wagmi's package](https://github.com/wevm/wagmi/blob/c862c126e7b8c93a97a368877d54d3acecfaa7fd/packages/react/test/index.tsx#L42) use a custom renderHook function, but I'm finding it very difficult to untangle the various imported types and methods.

**Comment by hardyjosh** [comment]
also curious - is nobody writing tests for their code?
